File: Queue/Queue.java
public class Queue {
    private Node lastNode;

    public Queue() {
        lastNode = null;
    }

    public boolean isEmpty() {
        return lastNode == null;
    }

    public void denqueueAll() {
        lastNode = null;
    }

    public void enqueue(int newItem) {
        Node newNode = new Node(newItem, lastNode);
        if(isEmpty()) {
            newNode.setNext(newNode);
        } else {
            newNode.setNext(lastNode.getNext());
            lastNode.setNext(newNode);
        }
        lastNode = newNode;
    }

    public int dequeue() {
        if (!isEmpty()) {
            Node firstNode = lastNode.getNext();
            if(firstNode == lastNode) {
                lastNode = null;
            } else {
                lastNode.setNext(firstNode.getNext());
            }
            return firstNode.getItem();
        } else return -1;
    }

    public int peek() {
        if(!isEmpty()) {
            Node firstNode = lastNode.getNext();
            return firstNode.getItem();
        } else return -1;
    }
}

File: Queue/Queue.c
#include "C:\\Users\\kritt\\Documents\\Personal_Project\\Data_Structure_and_Algorithm\\Linked_List\\Singly_Linked_List\\Node.h"
#include <stdio.h>

#define true 1
#define false 0
typedef int boolean;
struct Node *lastNode = NULL;

boolean isEmpty() {
    if(lastNode == NULL) {
        return true;
    } else return false;
}

void dequeueAll() {
    lastNode = NULL;
}

void enqueue(int newItem) {
    struct Node *newNode;
    newNode = createNode(newItem, NULL);
    if(isEmpty()) {
        setNext(newNode, newNode);
    } else {
        setNext(newNode, getNext(lastNode));
        setNext(lastNode, newNode);
    }
    lastNode = newNode;
}

int dequeue() {
    if(!isEmpty()) {
        struct Node *firstNode;
        firstNode = getNext(lastNode);
        if(firstNode == lastNode) {
            lastNode = NULL;
        } else {
            setNext(lastNode, getNext(firstNode));
        }
        return getItem(firstNode);
    } else return -1; 
}

int peek() {
    if(!isEmpty()) {
        struct Node *firstNode;
        firstNode = getNext(lastNode);
        return getItem(firstNode);
    } else return -1;
}

void main() {
    for(int i=0; i<9; i++) {
        enqueue(i);
    }
    while (!isEmpty())
    {
        printf("%d \n", dequeue());
    }
    
}